### PR TITLE
Update documentation regarding BokehJS API split

### PR DESCRIPTION
**Pull request by aerusso**
Addresses issues in #6184.

### CODE REVIEW

File: sphinx/source/docs/user_guide/bokehjs.rst [hunk 1]
```diff
@@ -337,3 +337,63 @@ function, with the plot it generates shown below:
...
+
+    <body>
+
+    <button onclick="addPoint()">Add some data!</button><br/>
+
+    <div>
+    <script type="text/javascript">
+
+    // arrays to hold data
+    source = new Bokeh.ColumnDataSource({
```
Comment (mattpap) [inline sphinx/source/docs/user_guide/bokehjs.rst]: `var source`. The same for `plot` and `scatterData`.

Comment (aerusso) [inline sphinx/source/docs/user_guide/bokehjs.rst]: Whoops! I tested with `use strict` now.

File: sphinx/source/docs/user_guide/bokehjs.rst [hunk 2]
```diff
@@ -337,3 +337,63 @@ function, with the plot it generates shown below:
...
+.. code-block:: html
+
+    <!doctype html>
+    <html lang="en">
+    <head>
+    <meta charset="utf-8">
+    <title>Complete Example</title>
+    <link rel="stylesheet" href="http://cdn.pydata.org/bokeh/release/bokeh-0.12.5.min.css" type="text/css" />
+    <script type="text/javascript" src="http://cdn.pydata.org/bokeh/release/bokeh-0.12.5.min.js"></script>
+    <script type="text/javascript" src="http://cdn.pydata.org/bokeh/release/bokeh-api-0.12.5.min.js"></script>
```
Comment (bryevdv) [inline sphinx/source/docs/user_guide/bokehjs.rst]: CDN links should be `https` (this will work but it's always better to demonstrate with `https`)

**Comment by bryevdv** [comment]
@aerusso This looks great, maybe for the short term a JSFiddle link or embed would be helpful to go along? I got your code running in this fiddle:

https://jsfiddle.net/bokeh/ck5udmxb/

I did have to change slightly, to split out the JS and HTML as jsfiddle prefers. Let me know what you think.

**Comment by bryevdv** [comment]
also I should add, for embedding (I think either the script or iframe would work), there is a raw HTML directive for sphinx:
```
.. raw:: html 

    [ html/css/js here ]

```

File: sphinx/source/docs/user_guide/bokehjs.rst (hunk 2 above)
Comment (aerusso) [inline sphinx/source/docs/user_guide/bokehjs.rst]: Thanks! I changed this.

**Comment by aerusso** [comment]
@bryevdv I actually ran into that "covering up other text" issue, but wasn't sure if I was just doing something wrong.

I don't know the proper way to add an html file into the Sphinx build process, but I implemented your suggestion with iframes in aerusso/bokeh@5eb9c8086842ed3d56292b842eb822e7633b2d29 . It builds with rst2html and does what you'd expect.

**Comment by bryevdv** [comment]
@aerusso Sorry I was pretty vague in my suggestion. I was not proposing to add an HTML file like that. If you click through the jsfiddle link above, there is a button to "embed" the fiddle. It gives you either s JS snippet, or an iframe, that you can drop into a page. I mean that you could use `raw:: html` with one of those things. Something like:

```
.. raw:: html 

    <script async src="//jsfiddle.net/bokeh/ck5udmxb/embed/"></script>
```

or 

```
.. raw:: html
 
    <iframe width="100%" height="300" src="//jsfiddle.net/bokeh/ck5udmxb/embedded/" allowfullscreen="allowfullscreen" frameborder="0"></iframe>
```

maybe with a note like "you can see this code live n the embedded JSFiddle below" or similar.

**Comment by bryevdv** [comment]
@mattpap can you comment on the "div covering up" issue? Does the user supplied div need to have `bokeh-root` class or something?

**Comment by aerusso** [comment]
I've added the JSFiddle, with a slightly larger height to make scrolling simpler.

**Comment by bryevdv** [comment]
FYI tests failing spuriously due to a downstream dependency changing

**Comment by bryevdv** [comment]
@aerusso It think this looks good to go, for my benefit can you attach a screenshot of the rendered docs page here?

**Comment by aerusso** [comment]
Attaching as requested. I built this by simply calling `rst2html` on `bokehjs.rst`, running a Python SimpleHTTPServer, and viewing the page through Google Chrome 58.

bokehjsapi-example-rendered

**Comment by bryevdv** [comment]
Thanks @aernlund ! Merging since the unrelated test issue is fixed in master.

**Pull request merged**